lld/ELF/ScriptParser.cpp

```diff
@@ -565,8 +565,17 @@ SmallVector<SectionCommand *, 0> ScriptParser::readOverlay() {
     addrExpr = readExpr();
     expect(":");
   }
-  Expr lmaExpr = consume("AT") ? readParenExpr() : Expr{};
-  expect("{");
+
+  Expr lmaExpr;
+  bool noCrossRefs = false;
+  while (auto tok = till("{")) {
```

Comment:
Would this permit AT to be used before NOCROSSREFS?

GNU ld documentation https://sourceware.org/binutils/docs/ld.html#Overlay-Description implies that it doesn't.
OVERLAY [start] : [NOCROSSREFS] [AT ( ldaddr )]

Empirically GNU ld accepts:
```
SECTIONS {
  OVERLAY : NOCROSSREFS AT(0x8000)  {
    .text { *(.text*) }
    .data { *(.data*) }
    .rodata { *(.rodata*) }
  }
}
```
But it won't accept
```
SECTIONS {
  OVERLAY : AT(0x8000) NOCROSSREFS   {
    .text { *(.text*) }
    .data { *(.data*) }
    .rodata { *(.rodata*) }
  }
}
```
aarch64-none-elf-ld:overlay.ld:2: syntax error

If I'm right about lld not caring about the order, then we'll accept scripts that GNU ld won't. May be worth sticking to what GNU ld accepts to help portability between toolchains.

Reply:
Good catch: I "zigged" when I should have "zagged". (Mistook the GNU LD grammar syntax.) Done.